src/tools/crossgen2/ILCompiler.DependencyAnalysisFramework/DependencyAnalyzer.cs

```diff
@@ -42,6 +42,31 @@ public sealed class DependencyAnalyzer<MarkStrategy, DependencyContextType> : De
         private bool _markingCompleted = false;
         private Random _stackPopRandomizer = null;
 
+        private class RandomInsertStack
```

Comment:
You could make this more generally useful (for what, I don't know) by making it generic, ie `RandomInsertStack<T>`, and instantiating it with `new RandomInsertStack<DependencyNodeCore<DependencyContextType>>` and replace all the `DependencyNodeCore<DependencyContextType>` instances with `T`.